src/Sarif.Sarifer/SariferOption.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+using Microsoft.VisualStudio.Shell;
+
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.CodeAnalysis.Sarif.Sarifer
+{
+    internal class SariferOption
+    {
+        private readonly AsyncPackage package;
+
+        private SariferExtensionOptionPage optionPage;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SariferOption"/> class.
+        /// Get visual studio option values
+        /// </summary>
+        /// <param name="package">Owner package, not null.</param>
+        private SariferOption(AsyncPackage package)
+        {
+            this.package = package ?? throw new ArgumentNullException(nameof(package));
+        }
+
+        public SariferExtensionOptionPage OptionPage
+        {
+            get
+            {
+                this.optionPage ??= (SariferExtensionOptionPage)this.package.GetDialogPage(typeof(SariferExtensionOptionPage));
+
+                return this.optionPage;
+            }
+        }
+
+
+        /// <summary>
+        /// Gets the instance of the command.
+        /// </summary>
+        public static SariferOption Instance
+        {
+            get;
+            private set;
+        }
+
+        public bool IsBackgroundAnalysisEnabled
+        {
+            get
+            {
+                try
+                {
+                    return this.OptionPage.BackgroundAnalysisEnabled;
+                }
+                catch
+                {
+                    // default value
+                    return true;
+                }
+            }
+        }
+
+        public bool IfAnalyzeSarifFile
```

Comment:
>IfAnalyzeSarifFile [](start = 20, length = 18)

change to ShouldAnalyzeSarifFile, or something similar

Reply:
updated

---
In reply to: [564968787](https://github.com/microsoft/sarif-visualstudio-extension/pull/354#discussion_r564968787) [](ancestors = 564968787)